libs/depth-chart/src/lib/hooks/use-depth-update.ts

```diff
@@ -0,0 +1,135 @@
+import type { ApolloError } from '@apollo/client';
+import { useApolloClient } from '@apollo/client';
+import throttle from 'lodash/throttle';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { updateDepthUpdate } from '../helpers';
+import {
+  MARKET_DEPTH_QUERY,
+  MARKET_DEPTH_UPDATE_SUB,
+} from '../queries/market-depth';
+import type {
+  marketDepth,
+  marketDepthVariables,
+} from '../__generated__/marketDepth';
+import type {
+  marketDepthUpdateSubscribe,
+  marketDepthUpdateSubscribeVariables,
+} from '../__generated__/marketDepthUpdateSubscribe';
+
+export interface QueryResult<TData> {
+  data: TData | undefined;
+  loading: boolean;
+  error?: ApolloError;
+}
+
+export function useDepthUpdate({ marketId }: marketDepthVariables, wait = 0) {
```

Comment:
We may want to switch to makeDataProvider
https://github.com/vegaprotocol/frontend-monorepo/blob/feature/151-orderbook/libs/orderbook/src/lib/marketdepth-data-provider.ts

Reply:
I've created #226 for this task.